### PR TITLE
Fix unicode

### PR DESCRIPTION
**Pull request by febeling**
Unicode strings weren't working and this PR fixes it. It's because `std::str::char_indices` yields byte offsets, not character indices. So I introduced two manually driven loop variables. Not sure if there's a nicer way?

### CODE REVIEW

**Comment by febeling** [comment]
@skade – Have an opinion?

**Comment by skade** [comment]
:+1: I was also surprised that it works with bot normalised and denormalised unicode.

**Comment by skade** [comment]
Wrong about the denormalised case. This fails:

``` rust
#[test]
fn unicode_denormalized() {
    assert_eq!(edit_distance::edit_distance("\u{0075}\u{0308}\u{006f}\u{0308}", "üö"), 0);
}
```

**Comment by febeling** [comment]
Yep. Nice catch! I didn't knew the whole concept, tbh. I generally stop with the snowman :stuck_out_tongue: 

Came up with this, and it also fails:

``` rust
#[test]
fn unicode_normalization() {
    assert_eq!(edit_distance::edit_distance("A\u{0308}ffin", "Äffin"), 0);
}
```

**Comment by febeling** [comment]
You think we should file a bug against `std::str::chars` for this?

**Comment by skade** [comment]
Nope, chars doesn't cover this case, it's raw unicode chars (without combining, etc.) only.

**Comment by febeling** [comment]
I think I'm gonna merge and consider normalization out of scope for this library :)

File: src/lib.rs [hunk 1]
```diff
@@ -46,8 +50,11 @@ pub fn edit_distance(a: &str, b: &str) -> i32 {
                 // match or substitution
                 matrix[i][j] + if ca == cb { 0 } else { 1 }];
             matrix[i+1][j+1] = *alternatives.iter().min().unwrap();
+            j = j + 1;
         }
+        j = 0;
+        i = i + 1;
```
Comment (febeling) [inline src/lib.rs:56]: @skade – I'm kinda unhappy about manually driving the loop counters here... You know any better way? `char_indices` would be just the thing, but the index is byte offset.

Comment (skade) [inline src/lib.rs:56]: Would [enumerate](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate) help? It's basically like rubys `each_with_index`.

Comment (febeling) [inline src/lib.rs:56]: That's exactly it! :pray:

Comment (febeling) [inline src/lib.rs:56]: Much better: 5df7da242f56a4f7878033d69f829130b2e43ecc

**Pull request merged**